File: sourcecode/Angular/client/src/app/_models/userParams.ts
import { User } from "./user";

export class UserParams{
    title: string;
    minAge = 18;
    maxAge = 99;
    pageNumber = 1;
    pageSize=5;
    orderBy ='lastActive';
    constructor(user:User){
        this.title = user.title === 'mr' ? 'mrs' : 'mr';
    }
}

File: sourcecode/aspnet-core-3-api/Helpers/UserParams.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WebApi.Helpers
{
    public class UserParams : PaginationParams
    {
        

        public int CurrentUserId { get; set; }

        public string Gender { get; set; }

        public int MinAge { get; set; } = 18;

        public int MaxAge { get; set; } = 165;

        public string OrderBy { get; set; } = "lastActive";
    }
}
